### PR TITLE
refactor(cesium): reorganize featureconverter

### PR DESCRIPTION
**Pull request by wallw-bits**
This is largely a reorganization, not a rewrite or a change in logic. `FeatureConverter` was a spaghetti mess and hard to make sense of for anyone outside of a couple of experts.

The general design for the refactor is shown in [`sync/iconverter.js`](https://github.com/ngageoint/opensphere/pull/853/files#diff-5a02209ea34ad23d4c9ae43b8590dd368) and [`sync/runconverter.js`](https://github.com/ngageoint/opensphere/pull/853/files#diff-17f72af38b0cd619696856187fcafa35). The individual converter implementations are all pulled in by [`sync/converter.js`](https://github.com/ngageoint/opensphere/pull/853/files#diff-07c1ba2319a19cbd682f725ad19ea7a9).

This is largely an attempt at feature-parity with the previous converter, with the following changes:

  - Several style bugs were fixed by consolidating and reusing style code where possible instead of using the previous one-off implementations
  - All functions being sent a `feature, geometry, style, context` now have a consistent parameter order everywhere
  - A few minor performance items were added (typically things like using scratch arrays/objects where possible)
  - All instances of `@suppress {checkTypes}` were removed and fixed
  - Most larger functions have been split into smaller blocks with a couple of exceptions
  - Cesium prefers wide scene trees rather than deep ones. Therefore, instead of having converters return a `Cesium.CollectionLike` of items (billboards, line primitives, polygon fill/outline primitves, polylines), we now only keep the root collections on the scene and keep track of the N items added per geometry.

I attempted to change anything outside of `FeatureConverter` the least amount possible (`VectorSynchronizer` and `VectorContext`) other than converting them to ES6 class syntax.

`FeatureConverter` was previously entirely missing tests. WebGL mocks have been added and are fleshed out enough that we can use a real Cesium scene in the tests. Many tests have been added but I still have more to write.

Note that for coverage reports to work on goog module files we need a change to the goog module preprocessor.

#### Basic Geometries
1. Load [geomtest.json.txt](https://github.com/ngageoint/opensphere/files/4119485/geomtest.json.txt) (point, multipoint, line, multiline, polygon, multipolygon, geometry collection)
    * Verify that the overall file looks the same in 2D and 3D (no features in one but not the other, etc.)
    * Verify that all the items change style properly when hovered/selected
    * Verify that the items are not shown when the layer's visibility is unchecked
    * Verify that all style items apply properly to each geometry (point/icon, color, fill color, line dash, size, opacity, etc.)
    * Verify that labels work on the geometry
1. Open the Timeline and use the Load range preset button to jump to this layer
    * Verify that it animates properly (especially with labels enabled)

#### Ellipses and Ellipsoids
1. Load [ellipse-test.txt](https://github.com/ngageoint/opensphere/files/4119498/ellipse-test.txt) (point, ellipse, ellipsoid)
    * Verify that the point appears at the origin
1. Change the style to Ellipse
    * Verify that the ellipse is displayed
    * Verify that all style items apply to the ellipse (size, color, fill color, etc.)
    * Verify that labels work on the ellipse
    * Repeat for other ellipse styles ("Ellipse with Center", etc.)
1. Check "Show Ground Reference"
    * Verify that the ground reference line is displayed
1. Check "Show Ellipsoids"
    * Verify that the ellipsoid is shown
    * Verify that the ellipsoid colors are correct for outline and fill (note: size is specifically ignored for ellipsoid outline width on master and I kept that)
    * Verify that labels work on the ellipsoid

##### Line String Tracks
1. Load [whale_shark.kml.txt](https://github.com/ngageoint/opensphere/files/4124832/whale_shark.kml.txt) (point/icon, line)
1. Uncheck the migration path at the bottom to get just the points
1. Draw a box and select the points
1. Right-click the layer and select "Create Track"
1. Ditch (or just hide) this layer to just view the track
    * Verify that the track displays properly with the line, endpoint, and label
    * Verify that all style items apply properly
1. Open the Timeline and use the Load preset button to jump to "Saved Places"
    * Animate over the track and verify that it displays as expected
    * Verify that all style items apply properly to each geometry (point/icon, color, fill color, line dash, size, opacity, etc.)

##### Multi Line String Tracks
1. Make two new points, one on either side of the antimeridian. Give each one a time instant (maybe back one up from now by a couple of hours for a good animation range).
1. Select both points
1. Right-click "Saved Places" and select "Create Track"
    * Verify that the track displays properly with the line, endpoint, and label
    * Verify that all style items apply properly
1. Open the Timeline and use the Load preset button to jump to "Saved Places"
    * Animate over the track and verify that it displays as expected
    * Verify that all style items apply properly

#### Altitude Mode
1. Enable Terrain
1. Load [AltitudeMode.kml.txt](https://github.com/ngageoint/opensphere/files/4300621/AltitudeMode.kml.txt)
1. Right-click the layer and select "Go To"
1. Points
    * Verify that "Mt. Everest Peak" is on the peak (clampToGround)
    * Verify that "Absolute Above Mt. Everest Peak" is 5km above the peak
    * Verify that "Relative to Mt. Everest Peak" is 2500m above the ground (above the peak; relativeToGround)
1. Lines
    * Verify that "Everest Line" follows the terrain, roughly through the peak (clampToGround)
    * Verify that "Absolute Everest Line" is at the same height as "Absolute Above Mt. Everest Peak"
    * Verify that "Relative Everest Line" does not work properly because Cesium does not support relativeToGround for lines
1. Polygons
    * Verify that "Everest Circle" follows the terrain, roughly centered on the peak (clampToGround)
    * Verify that "Absolute Everest Circle" is the same circle, but all at the altitude of "Absolute Above Mt. Everest Peak" 
    * Verify that "Relative to Everest Circle" does not work properly because Cesium does not support relativeToGround for lines/polygons.

#### Extrusions
1. Load [KML_Samples.kml.txt](https://github.com/ngageoint/opensphere/files/4300384/KML_Samples.kml.txt) (note: this is slightly modified from the original due to some stuff I noted in #907)
1. In KML Samples > Polygons > Extruded Polygon, right click the polygon and Go To it
    * Verify that it displays and highlights correctly
1. Turn on KML Samples > Paths and Go To that folder
    * Verify that each item displays and highlights correctly with the caveat that Cesium does not support `relativeToGround` for anything but points/icons (at least as far as I can tell) so anything marked "Relative" is really "Absolute" with the given altitude values.
    * Note: You will need to enable terrain to see the difference between the "Tessellated" and "Untessellated" lines

#### MultiPoint Volume
1. Load [multipoint_volume.json.txt](https://github.com/ngageoint/opensphere/files/4258717/multipoint_volume.json.txt)
    * Verify that the file performs well when loaded (the original complaint was ~2500 features with ~5500 total points was running at <10 fps on a high-quality machine)

### CODE REVIEW

**Comment by wallw-bits** [comment]
The test errors are coming from the Cesium 1.65 merge. I'll look into those.

**Comment by wallw-bits** [comment]
That test is failing intermittently for me locally too. I'll look at it.

File: src/plugin/file/csv/ui/csvimport.js [hunk 1]
```diff
@@ -11,6 +11,7 @@ goog.require('plugin.file.csv.CSVParserConfig');
 goog.require('plugin.file.csv.CSVProvider');
 
 
+// test comment
```
Comment (schmidtk) [inline src/plugin/file/csv/ui/csvimport.js]: Did the comment pass your test?

Comment (wallw-bits) [reply src/plugin/file/csv/ui/csvimport.js]: Indeed it did.

**Comment by schmidtk** [comment]
- When changing size or border on the layer style, polygons are not updated. Toggling to 2D and back to 3D will sync the style.
- When changing the icon on a layer style (ie, default to airplane), the initial icon size is larger than expected. Toggling to 2D and back to 3D will correct the size. This appears to only happen the first time an icon is selected for a particular layer. Subsequent uses of that icon init with the correct size.
Large Icons
- After the base icon size is fixed, the hover/selected icon size is still incorrect. It seems any color change to the icon uses the incorrect size.
Large Hover Icon
- In the debug build, drawing a polygon/circle/line (not box) triggers an assertion error:
```
Uncaught goog.asserts.AssertionError {message: "Failure: unhandled baseline undefined"}
goog.asserts.fail @ asserts.js:243
updateVerticalOrigin  @ labelconverter.js:239
```
- When creating a multi-track across the AM, I'm getting an error. I created four points alternating across the AM, starting on the left (+179) side. I can also reproduce by loading [MultiTrack.kml.txt](https://github.com/ngageoint/opensphere/files/4148469/MultiTrack.kml.txt) and hovering the line. The file contains a `gx:MultiTrack` comprised of two `gx:Track` elements.
```
multidynamiclinestringconverter.js:54 Uncaught TypeError: primitives.add is not a function
    at createOrUpdateDynamicMultiLineString (multidynamiclinestringconverter.js:54)
    at Object.update (multidynamiclinestringconverter.js:78)
    at runConverter (runconverter.js:20)
    at convertGeometry (converter.js:35)
    at convert (featureconverter.js:26)
    at VectorSynchronizer.updateFeature_ (vectorsynchronizer.js:458)
    at VectorSynchronizer.updateHighlightedItems_ (vectorsynchronizer.js:636)
    at VectorSynchronizer.onSourcePropertyChange_ (vectorsynchronizer.js:403)
    at plugin.file.kml.KMLSource.boundListener (events.js:17)
    at plugin.file.kml.KMLSource.ol.events.EventTarget.dispatchEvent (eventtarget.js:88)
```

**Comment by wallw-bits** [comment]
Most of those are fixed. I'm not totally happy with the icon stuff (there's a bit of a flicker to a larger size and back on rollover) and I'm continuing to try to improve that.

**Pull request merged**